### PR TITLE
Move logsumexp and pade from scipy.misc to scipy.special and scipy.interpolate

### PR DESCRIPTION
**Pull request by ev-br**
This is supposed to tick a box in the roadmap:

> scipy.misc will be removed as a public module. The functions in it can be moved to other modules:
> ...
> comb, factorial, logsumexp, pade : special

 `comb` and `factorial` have already been moved, this PR also moves `logsumexp`. EDIT:  and `pade`. 

One open question is whether `from scipy.misc import factorial, comb, logsumexp` should issue a DeprecationWarning pointing to their new location in `scipy.special`. At the moment, none of these does. If we want to actually remove `scipy.misc`, they probably should.

### CODE REVIEW

**Comment by person142** [comment]
Just wondering, but would it be better to put this stuff in separate files? (i.e. not in `basic.py`/`test_basic.py`.) Those files are IMO annoyingly large and unorganized.

File: scipy/special/__init__.py [hunk 1]
```diff
@@ -618,6 +618,7 @@
    round   -- Round to nearest integer
    xlogy   -- Compute ``x*log(y)`` so that the result is 0 if ``x = 0``.
    xlog1py -- Compute ``x*log1p(y)`` so that the result is 0 if ``x = 0``.
+   logsumexp  -- Compute the log of the sum of exponentials of input elements.
```
Comment (ev-br) [inline scipy/special/__init__.py]: Am I doing something wrong?

```
$ python refguide_summaries.py special
Traceback (most recent call last):
  File "refguide_summaries.py", line 112, in <module>
    main()
  File "refguide_summaries.py", line 79, in main
    doc = getattr(module, func).__doc__.split('\n')
AttributeError: 'module' object has no attribute 'roots_legendre'
```

Comment (person142) [inline scipy/special/__init__.py]: I'm at a loss... does `python -c "from scipy.special import roots_legendre` work?

**Comment by pv** [comment]
The script requires the development version of scipy installed in the 
current environment, which is usually not the case.
Probably works eg in python runtests.py --shell

File: scipy/special/__init__.py (hunk 1 above)
Comment (ev-br) [inline scipy/special/__init__.py]: No:

```
(scipy_py27)br@duneyrr:~/repos/scipy/tools$ python -c'from scipy.special import roots_legendre'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name roots_legendre 
```

But:

```
$ python runtests.py --ipython
Building, see build.log...
Build OK
Python 2.7.11 (default, Dec 14 2015, 22:56:59) 
Type "copyright", "credits" or "license" for more information.

IPython 4.1.2 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from scipy.special import roots_legendre

In [2]: roots_legendre?
Signature: roots_legendre(n, mu=False)
<snip>
```

Comment (ev-br) [inline scipy/special/__init__.py]: Indeed, `python setup.py install`, and then it works. Thanks for the heads-up, I missed the introduction of this helper.

**Comment by ev-br** [comment]
Comments addressed. Moved `logsumexp` to a separate file, squashed commits and force-pushed. A single commit is probably easier for git to follow so that the damage to git history is minimal.

**Comment by rgommers** [comment]
These changes look good to me. In addition this would be useful:
- a test to verify that ``misc.logsumexp`` exists and ``is special.logsumexp``.
- change the import in ``stats`` to import from ``special``

Regarding a deprecation or future warning: that could be annoying for users. They'd have to write version checks like ``if scipy.__version > '0.19.0': from scipy.special .... <etc>``. So what may be a better way is to migrate everything, give it another 2 releases or so, and then deprecate all of ``misc`` in one go.

**Comment by ev-br** [comment]
Comments addressed.
This PR also moves `pade`, for a good measure.

File: scipy/special/_pade.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    Returns
+    -------
+    p, q : Polynomial class
+        The pade approximation of the polynomial defined by `an` is
+        `p(x)/q(x)`.
+
+    Examples
+    --------
+    >>> from scipy import misc
```
Comment (rgommers) [inline scipy/special/_pade.py]: ``misc`` --> ``special``

Comment (ev-br) [inline scipy/special/_pade.py]: Nice catch! Fixed.

**Comment by person142** [comment]
Should `pade` really be in special? It seems like it would fit better in `interpolate`.

**Comment by ev-br** [comment]
Ok, redirected `pade` to `interpolate`, so that it's classed in "additional tools", along with `lagrange` and `taylor_polynomial`.

**Comment by person142** [comment]
LGTM.

File: scipy/misc/tests/test_common.py [hunk 3]
```diff
@@ -5,163 +5,17 @@
...
+    # scipy.special
+    assert_(logsumexp is sc_logsumexp)
 
-def test_logsumexp_b_shape():
-    a = np.zeros((4,1,2,1))
-    b = np.ones((3,1,5))
 
-    logsumexp(a, b=b)
+def test_pade():
+    assert(pade is i_pade)
```
Comment (rgommers) [inline scipy/misc/tests/test_common.py]: plain ``assert`` alert

**Comment by rgommers** [comment]
LGTM too, modulo the assert. Let's leave it open for a little longer, merge if there are no objections to putting ``pade`` in ``interpolate``.

File: scipy/misc/tests/test_common.py (hunk 3 above)
Comment (ev-br) [inline scipy/misc/tests/test_common.py]: Thanks, fixed

**Comment by rgommers** [comment]
Allright, in it goes. Thanks @ev-br !

**Pull request merged**